### PR TITLE
feat: extract function start candidates from .pdata

### PR DESCRIPTION
**Pull request by r0ny123**
- Implemented logic to extract function start candidates from `.pdata` section in `FunctionCandidateManager.py`.
- Iterates over `RUNTIME_FUNCTION` entries (12 bytes) and extracts the Start RVA.
- Corrected loop bounds to avoid out-of-bounds access.
- Removed TODO.
- Added regression test `tests/testPdataExtraction.py`.

### CODE REVIEW

File: tests/testPdataExtraction.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        self.base_addr = base_addr
+        self.binary = binary
+        self.code_areas = []
+
+    def getSections(self):
+        # yields name, start, end
+        # .pdata at 0x1000, size 36 (3 entries)
+        yield ".pdata", self.base_addr + 0x1000, self.base_addr + 0x1000 + 36
+
+class PdataExtractionTestSuite(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (r0ny123) [reply tests/testPdataExtraction.py]: > medium
> 
> The current test covers the happy path, but it would be beneficial to add more test cases to cover edge cases and ensure the implementation is robust. This is especially important given the potential bug in `FunctionCandidateManager.py`.
> 
> Consider adding tests for:
> 
> * A `.pdata` section that contains a zero entry, which should terminate parsing.
> * A `.pdata` section with a size that is not a multiple of the entry size (12 bytes), to ensure it's handled gracefully without errors.
> 
> Here's an example of a test for the zero-entry case that you could add to this test suite:
> 
> ```python
>     def test_pdata_with_zero_entry(self):
>         config = SmdaConfig()
>         fcm = FunctionCandidateManager(config)
> 
>         entries = [
>             (0x2000, 0x2040, 0x3000),
>             (0, 0, 0),  # Zero entry
>             (0x2100, 0x2150, 0x3020)
>         ]
> 
>         pdata_bytes = b''
>         for start, end, unwind in entries:
>             pdata_bytes += struct.pack('III', start, end, unwind)
> 
>         binary = bytearray(0x4000)
>         binary[0x1000 : 0x1000 + len(pdata_bytes)] = pdata_bytes
> 
>         disasm = DisassemblyResult()
>         disasm.binary_info = MockBinaryInfo(64, 0x140000000, bytes(binary), pdata_size=len(pdata_bytes))
>         
>         fcm.init(disasm)
> 
>         candidates = fcm.candidates
>         self.assertIn(0x140000000 + 0x2000, candidates)
>         self.assertNotIn(0x140000000 + 0x2100, candidates)
>         self.assertEqual(len(candidates), 1)
> ```

@jules

**Comment by r0ny123** [comment]
@codex review the entire PR carefully and suggest every possible improvements, bugs

**Comment by r0ny123** [comment]
@codex review

File: smda/intel/FunctionCandidateManager.py [hunk 2]
```diff
@@ -635,18 +635,18 @@ def locateStubChainCandidates(self):
...
-                        rva_function_candidate = None
-                        if len(packed_dword) == 4:
-                            rva_function_candidate = struct.unpack("I", packed_dword)[0]
-                            self.addExceptionCandidate(self.disassembly.binary_info.base_addr + rva_function_candidate)
-                        if not rva_function_candidate:
+                        if len(packed_dword) < 4:
                             break
+                        rva_function_candidate = struct.unpack("I", packed_dword)[0]
+                        if rva_function_candidate == 0:
+                            continue
```
[the remark this answers is not in the document]
Comment (r0ny123) [reply smda/intel/FunctionCandidateManager.py]: @codex fix this
